xarray/core/rolling.py

```diff
@@ -97,16 +89,55 @@ def __repr__(self):
         return "{klass} [{attrs}]".format(klass=self.__class__.__name__,
                                           attrs=','.join(attrs))
 
+    def __len__(self):
+        return len(self.obj[self.dim])
```

Comment:
Nit: prefer `self.obj.sizes[self.dim]` instead -- it avoids creating a DataArray object (which is somewhat more expensive).